cucumber-core/src/main/java/io/cucumber/core/options/PickleOrderParser.java

```diff
@@ -1,43 +1,38 @@
 package io.cucumber.core.options;
 
-import io.cucumber.core.logging.Logger;
-import io.cucumber.core.logging.LoggerFactory;
 import io.cucumber.core.order.PickleOrder;
-import io.cucumber.core.order.StandardPickleOrders;
+import io.cucumber.core.order.factory.PickleOrderFactory;
 
-import java.util.Random;
+import java.util.Optional;
+import java.util.ServiceLoader;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 final class PickleOrderParser {
 
-    private static final Logger log = LoggerFactory.getLogger(PickleOrderParser.class);
-
-    private static final Pattern RANDOM_AND_SEED_PATTERN = Pattern.compile("random(?::(\\d+))?");
+    private static final Pattern NAME_WITH_ARGUMENT_PATTERN = Pattern.compile("([a-z_]+)(?::(.*))?");
 
     static PickleOrder parse(String argument) {
-        if ("reverse".equals(argument)) {
-            return StandardPickleOrders.reverseLexicalUriOrder();
-        }
-
-        if ("lexical".equals(argument)) {
-            return StandardPickleOrders.lexicalUriOrder();
-        }
-
-        Matcher matcher = RANDOM_AND_SEED_PATTERN.matcher(argument);
-        if (!matcher.matches()) {
-            throw new IllegalArgumentException("Invalid order. Must be either reverse, random or random:<long>");
+        Matcher matcher = NAME_WITH_ARGUMENT_PATTERN.matcher(argument);
+        if (matcher.matches()) {
+            return parse(matcher.group(1), matcher.group(2));
         }
+        return parse(argument, null);
```

Comment:
This does need a test. The others can be ignored.

Reply:
Okay, I created specific class for RandomPickleOrder to make it easier to test, that the mechanism returns this kind of order. I like concrete classes rather than anonymous lambdas for this reason :)

This documentation is a bit confusing since it's spread so much. I made some effort.

Reply:
That's fine. As long as they're package private.

Reply:
The USAGE.txt is still a bit of an embarrassment. I'd love to replace it with something like Pico CLI but that's a long term project that requires fixing quite a few of Cucumbers internal first.

But for the other docs, that is the unfortunate nature of documentation. It's not so bad if you keep in mind that the README's are for users while the Javadoc is for developers. You have to explain somethings twice, but you can keep the details in the Javadoc.